jobs/payment-jobs/tasks/ejv_payment_task.py

```diff
@@ -215,25 +202,38 @@ def _create_ejv_file_for_gov_account(cls, batch_type: str):  # pylint:disable=to
 
             current_app.logger.info("Creating ejv invoice link records and setting invoice status.")
             sequence = 1
+
             for inv in invoices:
-                current_app.logger.debug(f"Creating EJV Invoice Link for invoice id: {inv.id}")
-                ejv_invoice_link = EjvLinkModel(
-                    link_id=inv.id,
-                    link_type=EJVLinkType.INVOICE.value,
-                    ejv_header_id=ejv_header_model.id,
-                    disbursement_status_code=DisbursementStatus.UPLOADED.value,
-                    sequence=sequence,
-                )
-                db.session.add(ejv_invoice_link)
-                sequence += 1
-                current_app.logger.debug(f"Creating Invoice Reference for invoice id: {inv.id}")
-                inv_ref = InvoiceReferenceModel(
-                    invoice_id=inv.id,
-                    invoice_number=generate_transaction_number(inv.id),
-                    reference_number=None,
-                    status_code=InvoiceReferenceStatus.ACTIVE.value,
-                )
-                db.session.add(inv_ref)
+                if cls._is_partial_refund(inv):
+                    db.session.add_all([
+                        EjvLinkModel(
+                            link_id=pr.id,
+                            link_type=EJVLinkType.PARTIAL_REFUND.value,
+                            ejv_header_id=ejv_header_model.id,
+                            disbursement_status_code=DisbursementStatus.UPLOADED.value,
+                            sequence=seq,
+                        )
+                        for seq, pr in enumerate(partial_refunds, start=sequence)
+                    ])
+                    inv.refund_date = datetime.now(timezone.utc)
```

Comment:
refund_date helps get_partial_refunds_invoices avoid adding duplicate partial_refunds.
![image](https://github.com/user-attachments/assets/1a48de70-92c9-405a-b26f-72d3c717c38f)

Reply:
shouldn't we just check partial_refunds table to see if it exists for the invoice?

Just not sure if refund_date is set correctly in every situation

Reply:
that's a good idea! I was thinking refund_date suppose to be set by reconciliation, not payment-jobs, it could fail

Reply:
With the status we'd know where it failed roughly

Reply:
from current MVP site, it only checked if invoice.refund > 0 -> there are existing partial refunds
a bit different here, I want to avoid the duplications, I'll add a status for the partial_refund.